app/validators/embargo_date_parts.rb

```diff
@@ -0,0 +1,14 @@
+# typed: true
+# frozen_string_literal: true
+
+# Validates embargo month and day
+class EmbargoDateParts < ActiveModel::Validator
+  def validate(record)
+    return if record.send('embargo_date').instance_of?(Date)
```

Comment:
Can you explain why this clause is necessary?

Reply:
Also, why `send` and not just `record.embargo_date`?

Reply:
Thanks, changed to  use `record.embargo_date`. The reason for this guard clause is that this test https://github.com/sul-dlss/happy-heron/blob/1190a978784dc06e275873c6970eceb5cededbcd/spec/requests/create_work_spec.rb#L261, fails because a valid Date is being passed into the `EmbargoDateParts` validator and we removed the month and day parameters in the filter https://github.com/sul-dlss/happy-heron/blob/1190a978784dc06e275873c6970eceb5cededbcd/app/forms/embargo_date_params_filter.rb#L20-L21
causing the validation to fail as we would expect. Both this guard clause and the next line are because of test failures (see also my comment below).

Reply:
Can you add a comment to the effect of, `# If we've already successfully deserialized the date, there's no point to continue with this`